### PR TITLE
Support websocket in `createAndExport` function

### PR DESCRIPTION
**Pull request by npty**
# Description
Support WebSocket in `createAndExport` function. the caller must set the `ws`  flag to `true` when passing the options object to the function.

## Usage

To start the WebSocket server, specify `ws: true` to the object that passes into `createAndExport` function.
```ts
import {
  createAndExport,
} from "@axelar-network/axelar-local-dev";

(async () => {
  await createAndExport({ws: true});
})()
```

The websocket server URL is exposed to `local.json` file. here's an example.

```json
[
  {
    "name": "Moonbeam",
    "chainId": 2500,
    "gateway": "0xD9C3b5474f1ec7578E3549BAd3ce266bAa5c2D3c",
    "gasReceiver": "0x5737A424F8e9A0dbA8701A4cDaB8E425F9031b33",
    "constAddressDeployer": "0x69aeB7Dc4f2A86873Dae8D753DE89326Cf90a77a",
    "rpc": "http://localhost:8500",
    "ws": "ws://localhost:8500",
    "tokenName": "DEV",
    "tokenSymbol": "DEV"
  },
  {
    "name": "Avalanche",
    "chainId": 2501,
    "gateway": "0x411fB36E488f22bBE872C345e652e45ff0EB319f",
    "gasReceiver": "0xc17a3E5357146d7b9CC3fB025356f25dd450554F",
    "constAddressDeployer": "0x69aeB7Dc4f2A86873Dae8D753DE89326Cf90a77a",
    "rpc": "http://localhost:8501",
    "ws": "ws://localhost:8501",
    "tokenName": "Avax",
    "tokenSymbol": "AVAX"
  }
]
```

### Check if the WebSocket works

1. Connect to the websocket endpoint
```bash
wscat --connect ws://localhost:8500
Connected (press CTRL+C to quit)
>
```

2. Starts the local dev server

```ts
import { createAndExport } from "@axelar-network/axelar-local-dev";
const pk = process.env.pk;
const wallet = new ethers.Wallet(pk);
const callback = async (network: any) => {
  await network.deployToken("Axelar Wrapped aUSDC", "aUSDC", 6, BigInt(1e70));
  await network.giveToken(wallet.address, "aUSDC", BigInt(1e70));
};

await createAndExport({
  ws: true,
  accountsToFund: [wallet.address],
  chains: ["Moonbeam", "Avalanche"],
  callback,
});
```

3. Subscribe to the `Transfer` event

```bash
wscat --connect ws://localhost:8500
Connected (press CTRL+C to quit)
> {"jsonrpc":"2.0","id":1,"method":"eth_subscribe","params":["logs", {"address": "0xdeD4Dfb12796288F855D590e4750bB0ea17DC421", "topics": ["0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef"]}]}
< {"id":1,"jsonrpc":"2.0","result":"0x1"}
```

4. Send a `Transfer` Tx

```ts
const pk = process.env.pk
const chains = fs.readFileSync("./local.json", "utf8");
const chain = JSON.parse(chains)[0];
const wallet = new ethers.Wallet(
  pk,
  new ethers.providers.JsonRpcProvider(chain.rpc)
);
const contractAddress = "0xdeD4Dfb12796288F855D590e4750bB0ea17DC421";
const contract = new ethers.Contract(
  contractAddress,
  [
    "function transfer(address to, uint256 amount)",
  ],
  wallet
);

const tx = await contract
  .transfer("0x5737A424F8e9A0dbA8701A4cDaB8E425F9031b33", 100000)
  .then((tx: any) => tx.wait());
```

5. Receive an event

```bash
wscat --connect ws://localhost:8500
Connected (press CTRL+C to quit)
> {"jsonrpc":"2.0","id":1,"method":"eth_subscribe","params":["logs", {"address": "0xdeD4Dfb12796288F855D590e4750bB0ea17DC421", "topics": ["0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef"]}]}
< {"id":1,"jsonrpc":"2.0","result":"0x1"}
< {"jsonrpc":"2.0","method":"eth_subscription","params":{"result":{"address":"0xded4dfb12796288f855d590e4750bb0ea17dc421","blockHash":"0x75ff5c08d3fb351f01fc9889edfaaee4bfe93cfc855c1521b5579703cbac1ff1","blockNumber":"0xe","data":"0x00000000000000000000000000000000000000000000000000000000000186a0","logIndex":"0x0","removed":false,"topics":["0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef","0x000000000000000000000000c1f0ceac1116a727a1893248ed13b9a0f890a350","0x0000000000000000000000005737a424f8e9a0dba8701a4cdab8e425f9031b33"],"transactionHash":"0x0c66556651e74ba95861f1bd3ef6c5558f624978e50c46613e018796e88ec281","transactionIndex":"0x0"},"subscription":"0x1"}}
```

### Check if it breaks the `axelar-local-gmp-examples` repo

Run `npm run test`

image

### CODE REVIEW

**Comment by npty** [comment]
Published version `1.3.0-beta.0` to npm

File: src/exportUtils.ts [hunk 1]
```diff
@@ -42,46 +43,63 @@ export async function createAndExport(options: CreateLocalOptions = {}) {
         chains: ['Moonbeam', 'Avalanche', 'Fantom', 'Ethereum', 'Polygon'],
         port: 8500,
         relayInterval: 2000,
-    } as CreateLocalOptions;
-    for (const option in defaultOptions) (options as any)[option] = (options as any)[option] || (defaultOptions as any)[option];
-    const chains_local: Record<string, any>[] = [];
-    let i = 0;
-    for (const name of options.chains!) {
+    };
+    const _options = { ...defaultOptions, ...options };
```
Comment (StephenFluin) [inline src/exportUtils.ts:47]: Why the underscore here?

File: src/networkUtils.ts [hunk 2]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
```
Comment (StephenFluin) [inline src/networkUtils.ts:1]: Why disable this? No requires seems like a good idea.

**Review by StephenFluin — APPROVED** [review]
couple nits / questions, but LGTM

**Pull request closed without merging**